File: examples/ruby/solarfox-api-request.rb
require 'net/http'
require 'json'
require 'uri'
require 'time'

uri = URI.parse('http://api.solar-fox.com/import')
headers = {'Content-Type' => 'application/json', 'Authorization' => 'test'}
data = {
  Headquarters: {
    "WR-118412E": {
      type: "ELECTRICITY_CONSUMPTION_ENERGY_ABSOLUTE",
      readings: [
        {
          timestamp: Time.now.to_i,
          value: 128013.0
        }
      ]
    }
  }
}.to_json

# Prepare request
http = Net::HTTP.new(uri.host,uri.port)
req = Net::HTTP::Post.new(uri.path, initheader = headers)
req.body = data

# Execute
begin
  res = http.request(req)
rescue Exception => e
  print "Error: " + e
else
  # Collect headers
  res_headers = []
  res.each_header do |k, v|
    res_headers << "  #{k}: #{v}"
  end

  puts "Status: " + res.code
  puts "Headers:", res_headers
  puts "Body: " + res.body
end

File: examples/node/solarfox-api-request.js
var http = require('http');

var postData = JSON.stringify({
  "Headquarters": {
    "WR-118412E": {
      "type": "ELECTRICITY_CONSUMPTION_ENERGY_ABSOLUTE",
      "readings": [
        {
          "timestamp": Math.floor(Date.now() / 1000),
          "value": 128013.0
        }
      ]
    }
  }
});

var options = {
  hostname: 'api.solar-fox.com',
  port: 80,
  path: '/import',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Authorization': 'test'
  }
};

var req = http.request(options, function(res) {
  console.log('Status:', res.statusCode);
  console.log('Headers:', JSON.stringify(res.headers));
  res.setEncoding('utf8');
  res.on('data', function (chunk) {
    console.log('Body:', chunk);
  });
});

req.on('error', function(e) {
  console.error('Error:', e.message);
});

req.write(postData);
req.end();
